dashboard/src/reducers/kube.ts

```diff
@@ -28,6 +30,37 @@ const kubeReducer = (
         [action.payload.key]: { isFetching: false, error: action.payload.error },
       };
       return { ...state, items: { ...state.items, ...erroredItem } };
+    case getType(actions.kube.openWatchResource):
+      const { ref, handler } = action.payload;
+      key = ref.watchResourceURL();
+      if (state.sockets[key]) {
+        // Socket for this resource already open, do nothing
+        return state;
+      }
+      const socket = ref.watchResource();
+      socket.addEventListener("message", handler);
+      return {
+        ...state,
+        sockets: {
+          ...state.sockets,
+          [key]: socket,
+        },
+      };
+    // TODO(adnan): this won't handle cases where one component closes a socket
+    // another one is using. Whilst not a problem today, a reference counter
+    // approach could be used here to enable this in the future.
+    case getType(actions.kube.closeWatchResource):
+      key = action.payload.watchResourceURL();
+      const { sockets } = state;
+      const { [key]: foundSocket, ...restSockets } = sockets;
```

Comment:
:ok_hand: that's neat

Minor thing, I would change `restSockets` for `otherSockets` just to make it clear it's not a RESTful thing

Reply:
indeed!

I don't have a preference so happy to change it, but just for your reference, the term rest here comes from this being the rest operator. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#Rest_syntax_(parameters) and https://underscorejs.org/#rest.

Reply:
okay, it's just the first thing I thought when I read `restSockets`